_examples/helloworld/helloworld_integration_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 ETH Zurich
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build integration
+
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/netsec-ethz/scion-apps/pkg/integration"
+)
+
+const (
+	name = "helloworld"
+	cmd  = "helloworld"
+)
+
+func TestHelloworldSample(t *testing.T) {
+	if err := integration.Init(name); err != nil {
+		t.Fatalf("Failed to init: %s\n", err)
+	}
+	// Common arguments
+	cmnArgs := []string{}
+	// Server
+	serverPort := "12345"
+	serverArgs := []string{"-port", serverPort}
+	serverArgs = append(serverArgs, cmnArgs...)
+
+	testCases := []struct {
+		Name              string
+		Args              []string
+		ServerOutMatchFun func(bool, string) bool
+		ServerErrMatchFun func(bool, string) bool
+		ClientOutMatchFun func(bool, string) bool
+		ClientErrMatchFun func(bool, string) bool
```

Comment:
Could the framework just call these matchers on the full output instead? That would allow to get rid of the `bool` parameter, which I find a bit surprising.
Note, perhaps, that the golang regexp has a multi-line mode so that matches for a full line are still easily possible -- it's enabled with a `(?m)` in the regexp string, e.g.:
```
`(?m)^foo$`
```